middle_end/flambda2.0/to_cmm/un_cps_static.ml

```diff
@@ -387,23 +396,36 @@ let static_const
     env r ~params_and_body
     (bound_symbols : Bound_symbols.t)
     (static_const : Static_const.t) =
-  (* Gc roots: statically allocated blocks themselves do not need to be scanned,
-     however if statically allocated blocks contain dynamically allocated
-     contents, then that block has to be registered as Gc roots for the Gc to
-     correctly patch it if/when it moves some of the dynamically allocated
-     blocks. As a safe over-approximation, we thus register as gc_roots all
-     symbols who have an associated computation (and thus are not
-     fully_static). *)
-  let roots =
-    if Static_const.is_fully_static static_const then []
-    else Symbol.Set.elements (Bound_symbols.being_defined bound_symbols)
-  in
-  let r = R.add_gc_roots r roots in
-  let env, r, update_opt =
-    static_const0 env r ~params_and_body bound_symbols static_const
-  in
-  (* [R.archive_data] helps keep definitions of separate symbols in different
-     [data_item] lists and this increases readability of the generated Cmm. *)
-  env, R.archive_data r, update_opt
-
+  try
+    (* Gc roots: statically allocated blocks themselves do not need to be scanned,
+       however if statically allocated blocks contain dynamically allocated
+       contents, then that block has to be registered as Gc roots for the Gc to
+       correctly patch it if/when it moves some of the dynamically allocated
+       blocks. As a safe over-approximation, we thus register as gc_roots all
+       symbols who have an associated computation (and thus are not
+       fully_static). *)
+    let roots =
+      if Static_const.is_fully_static static_const then []
+      else Symbol.Set.elements (Bound_symbols.being_defined bound_symbols)
+    in
+    let r = R.add_gc_roots r roots in
+    let env, r, update_opt =
+      static_const0 env r ~params_and_body bound_symbols static_const
+    in
+    (* [R.archive_data] helps keep definitions of separate symbols in different
+       [data_item] lists and this increases readability of the generated Cmm. *)
+    env, R.archive_data r, update_opt
+  with Misc.Fatal_error as e ->
```

Comment:
You should guard this with `if !Clflags.flambda2_context_on_error`, I think

Reply:
this is done in the last commit